sphinx_doc/tutorials/docs/writing_new_nav2planner_plugin.rst

```diff
@@ -0,0 +1,196 @@
+
+For this tutorial, we will be using methods ``RRTConnect::configure()`` and ``RRTConnect::createPlan()`` to create RRT Connect planner.
+
+In our planner, ``RRTConnect::configure()`` method sets member variables and ROS parameters,
+
+.. code-block:: c
+
+  node_ = parent;
+  tf_ = tf;
+  name_ = name;
+  costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+
+  // Parameter initialization
+  nav2_util::declare_parameter_if_not_declared(node_, name_ + ".solve_time", rclcpp::ParameterValue(1.0));
+  node_->get_parameter(name_ + ".solve_time", solve_time_);
+  nav2_util::declare_parameter_if_not_declared(node_, name_ + ".max_trials", rclcpp::ParameterValue(4));
+  node_->get_parameter(name_ + ".max_trials", max_trials_);
+  nav2_util::declare_parameter_if_not_declared(node_, name_ + ".collision_checking_resolution", rclcpp::ParameterValue(0.001));
+  node_->get_parameter(name_ + ".collision_checking_resolution", collision_checking_resolution_);
+  nav2_util::declare_parameter_if_not_declared(node_, name_ + ".allow_unknown", rclcpp::ParameterValue(true));
+  node_->get_parameter(name_ + ".allow_unknown", allow_unknown_);
+
+and also, it sets OMPL variables such as bounds, setup settings and space validation.
+
+In ``RRTConnect::createPlan()`` method, RRT Connect planner is called using start pose and goal pose to solve the global path planning problem. Upon succeding, it converts the path to the ``nav_msgs::msg::Path`` and returns to the caller which is planner server instance.
+
+The remaining methods are not used but its mandatory to override them. As per the rules, we did override all but left them blank.
+
+2- Exporting the planner plugin
+-------------------------------
+
+Now that we have created our custom planner, we need to export our planner plugin so that it would be visible to the planner server. Plugins are loaded at runtime and if they are not visible, then our planner server won't be able to load it. In ROS2, exporting and loading plugins is handled by ``pluginlib``.
+
+Coming to our tutorial, class ``nav2_rrtconnect_planner::RRTConnect`` is loaded dynamically as ``nav2_core::GlobalPlanner`` which is our base class.
+
+1. To export the planner, we need to provide two lines
+
+.. code-block:: c
+  
+  #include "pluginlib/class_list_macros.hpp"
+  PLUGINLIB_EXPORT_CLASS(nav2_rrtconnect_planner::RRTConnect, nav2_core::GlobalPlanner)
+
+Note that it requires pluginlib to export out plugin's class. Pluginlib would provide as macro ``PLUGINLIB_EXPORT_CLASS`` which does all the work of exporting.
+
+It is good practice to place this lines at the end of the file but in practice you can also write at the top.
+
+2. Next step would be to create plugin's description file in the root directory of the package. For example, ``global_planner_plugin.xml`` file in our tutorial package. This file contains following information
```

Comment:
Is this information in the costmap tutorial? If not, can you add it? Its good context and this section is similar for both so ideally, I'd like them to be basically identical, minus the context / names.

Reply:
Sure!